### ISSUE TITLE
Could not locate renter for offer cause of insufficient FIND_NODE

### ISSUE THREAD

**Issue by littleskunk** [Storj/core#182]
### Package Versions

Replace the values below using the output from `npm list storj`.

```
/usr/local/lib
└─┬ storjshare-cli@4.0.2 (git://github.com/littleskunk/storjshare-cli.git#698482a90f53845f43b98a1226d277058aebc18c)
  └─┬ storj@1.2.5
    └── kad@1.5.12
```
Note: Open pull request. I added configuration for MAX_CONCURRENT_OFFERS. This was needed to reprocude this error.

Replace the values below using the output from `node --version`.

```
v4.4.5
```

### Expected Behavior

Please describe the program's expected behavior. Include an example of your
usage code in the back ticks below if applicable.

```
I would like to locate a renter.
Send out a few FIND_NODE and wait for the response. Take the node closest to the renter and repeat it until the closest node is found. The closest node should be the renter.
```

### Actual Behavior

Please describe the program's actual behavior. Please include any stack traces
or log output in the back ticks below.

I have 20 nodes on my shortlist and if I am lucky each of them can guide me to 20 other nodes. The more nodes the network has the more `Could not locate contract for offer` I will see because the implementation gives me only access to 420 nodes. I would expect the next FIND_NODE round with the closest node out of the 420 nodes. That would reduce the `Could not locate contract for offer` to less than 1%. At the moment we have ~10%

MAX_CONCURRENT_OFFERS is set to 1 to avoid overlapping offers. There is only 1 active offer that will send out 3 FIND_NODE messages. They returned 48 nodes. One of the 48 nodes must be closer to the renter but I did not start a new FIND_NODE round to ask them.
```
{"level":"debug","message":"Received contract offer...","timestamp":"2016-06-16T12:29:52.609Z"}
{"level":"debug","message":"pending offers is less than concurrency: true","timestamp":"2016-06-16T12:29:52.612Z"}
{"level":"debug","message":"negotiator returned: true","timestamp":"2016-06-16T12:29:52.612Z"}
{"level":"debug","message":"we have enough free space: true","timestamp":"2016-06-16T12:29:52.612Z"}
{"level":"info","message":"sending PUBLISH message to {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080192698}","timestamp":"2016-06-16T1
2:29:52.698Z"}
{"level":"debug","message":"queuing callback for reponse to 8bc46e55190e59a8a570d7889a2a74ed9ee68068","timestamp":"2016-06-16T12:29:52.757Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080192758}","timestamp":"2016-06-16T1
2:29:52.758Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080192758}","timestamp":"2016-06-16T12:29:52.75
9Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:52.759Z"}
{"level":"debug","message":"Received contract offer...","timestamp":"2016-06-16T12:29:52.759Z"}
{"level":"debug","message":"pending offers is less than concurrency: false","timestamp":"2016-06-16T12:29:52.762Z"}
{"level":"debug","message":"negotiator returned: true","timestamp":"2016-06-16T12:29:52.762Z"}
{"level":"debug","message":"we have enough free space: true","timestamp":"2016-06-16T12:29:52.762Z"}
{"level":"debug","message":"Not taking the contract...","timestamp":"2016-06-16T12:29:52.762Z"}
{"level":"info","message":"sending PUBLISH message to {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080192762}","timestamp":"2016-06-16T1
2:29:52.762Z"}
{"level":"debug","message":"queuing callback for reponse to 177f311cca674e23cbe47a5a400c9d4e1ce2b969","timestamp":"2016-06-16T12:29:52.822Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080192824}","timestamp":"2016-06-16T1
2:29:52.824Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080192824}","timestamp":"2016-06-16T12:29:52.82
4Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:52.825Z"}
{"level":"info","message":"replying to message to 20fef648afeb97ee24ad59724acb96379408e23f","timestamp":"2016-06-16T12:29:52.825Z"}
{"level":"debug","message":"not waiting on callback for message 20fef648afeb97ee24ad59724acb96379408e23f","timestamp":"2016-06-16T12:29:52.885Z"}
{"level":"debug","message":"searching for nodes close to key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:54.385Z"}
{"level":"debug","message":"performing network walk for NODE c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:54.386Z"}
{"level":"debug","message":"starting contact iteration for key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:54.386Z"}
{"level":"debug","message":"querying 6ad8a5a30528b3572f3e80cbd5a3eb4c52b8b8ac for key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:54.387Z"}
{"level":"info","message":"sending FIND_NODE message to {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6000,\"nodeID\":\"6ad8a5a30528b3572f3e80cbd5a3eb4c52b8b8ac\",\"lastSeen\":1466080194387}","timestamp":"2016-06
-16T12:29:54.387Z"}
{"level":"debug","message":"queuing callback for reponse to 4d22b95e38fe5579330e841715a5242d22c7f254","timestamp":"2016-06-16T12:29:54.419Z"}
{"level":"debug","message":"querying 1495a38adfec7117a2dc84135be8ea9612841a86 for key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:54.420Z"}
{"level":"info","message":"sending FIND_NODE message to {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194420}","timestamp":"2016-06-16
T12:29:54.420Z"}
{"level":"debug","message":"queuing callback for reponse to 6fded0734b113d64cea6a0e1864fbc1593338b52","timestamp":"2016-06-16T12:29:54.480Z"}
{"level":"debug","message":"querying ef409f8911a486759f59276ad83626bdb0fd97d4 for key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:54.481Z"}
{"level":"info","message":"sending FIND_NODE message to {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080194481}","timestamp":"2016-06-16T12:29:54.481Z"}
{"level":"debug","message":"queuing callback for reponse to a111b7dd674abfc4950fd29eac0b95bebb721f38","timestamp":"2016-06-16T12:29:54.554Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194563}","timestamp":"2016-06-16T12:29:54.563Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194563}","timestamp":"2016-06-16T12:29:54.563Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:54.563Z"}
{"level":"debug","message":"checking pending rpc callback stack for 2eb1c56a71e95a71f3afb668ca47f74aa09c8572","timestamp":"2016-06-16T12:29:54.632Z"}
{"level":"info","message":"replying to message to c26348a3dfdcd9ff3ae8b4c8ed0ad66820324069","timestamp":"2016-06-16T12:29:54.633Z"}
{"level":"debug","message":"not waiting on callback for message c26348a3dfdcd9ff3ae8b4c8ed0ad66820324069","timestamp":"2016-06-16T12:29:54.666Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6155,\"nodeID\":\"2b6b9175dcbdcbbce3337ca9e7a73946b82452ea\",\"lastSeen\":1466080194667}","timestamp":"2016-06-16T12:29:54.667Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6155,\"nodeID\":\"2b6b9175dcbdcbbce3337ca9e7a73946b82452ea\",\"lastSeen\":1466080194667}","timestamp":"2016-06-16T12:29:54.667Z"}
{"level":"debug","message":"contact not in bucket, moving to head","timestamp":"2016-06-16T12:29:54.668Z"}
{"level":"debug","message":"checking pending rpc callback stack for 1648962f0513b018a14633ece889a7b9f300d73b","timestamp":"2016-06-16T12:29:54.736Z"}
{"level":"warn","message":"dropping received late response to 1648962f0513b018a14633ece889a7b9f300d73b","timestamp":"2016-06-16T12:29:54.736Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194737}","timestamp":"2016-06-16T12:29:54.737Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194737}","timestamp":"2016-06-16T12:29:54.737Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:54.737Z"}
{"level":"debug","message":"checking pending rpc callback stack for fa9faa7585f33cf1087756fa04246c89d0724f47","timestamp":"2016-06-16T12:29:54.807Z"}
{"level":"info","message":"replying to message to a6d18bbe87a8f05ffd15a1c3b5ce806e597ae4b0","timestamp":"2016-06-16T12:29:54.807Z"}
{"level":"debug","message":"not waiting on callback for message a6d18bbe87a8f05ffd15a1c3b5ce806e597ae4b0","timestamp":"2016-06-16T12:29:54.839Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194840}","timestamp":"2016-06-16T12:29:54.841Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194840}","timestamp":"2016-06-16T12:29:54.841Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:54.841Z"}
{"level":"debug","message":"checking pending rpc callback stack for 369d9ef4c4138fc08b099df43d10a44742f18a15","timestamp":"2016-06-16T12:29:54.908Z"}
{"level":"info","message":"replying to message to 69f6e76c21bbdb656ac08a3898f74140352ab648","timestamp":"2016-06-16T12:29:54.908Z"}
{"level":"debug","message":"not waiting on callback for message 69f6e76c21bbdb656ac08a3898f74140352ab648","timestamp":"2016-06-16T12:29:54.941Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194942}","timestamp":"2016-06-16T12:29:54.942Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194942}","timestamp":"2016-06-16T12:29:54.942Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:54.942Z"}
{"level":"debug","message":"checking pending rpc callback stack for 8bc46e55190e59a8a570d7889a2a74ed9ee68068","timestamp":"2016-06-16T12:29:55.012Z"}
{"level":"info","message":"replying to message to 297ff597f9672c9b1f7ce6a7688f36cba179c41c","timestamp":"2016-06-16T12:29:55.012Z"}
{"level":"debug","message":"not waiting on callback for message 297ff597f9672c9b1f7ce6a7688f36cba179c41c","timestamp":"2016-06-16T12:29:55.044Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080195045}","timestamp":"2016-06-16T12:29:55.046Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080195045}","timestamp":"2016-06-16T12:29:55.046Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.046Z"}
{"level":"debug","message":"checking pending rpc callback stack for 177f311cca674e23cbe47a5a400c9d4e1ce2b969","timestamp":"2016-06-16T12:29:55.113Z"}
{"level":"info","message":"replying to message to 325ed1a6f7c16222e352053386798e403d64b98f","timestamp":"2016-06-16T12:29:55.114Z"}
{"level":"debug","message":"not waiting on callback for message 325ed1a6f7c16222e352053386798e403d64b98f","timestamp":"2016-06-16T12:29:55.173Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080195180}","timestamp":"2016-06-16T12:29:55.180Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080195180}","timestamp":"2016-06-16T12:29:55.180Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.180Z"}
{"level":"debug","message":"Received contract offer...","timestamp":"2016-06-16T12:29:55.181Z"}
{"level":"debug","message":"pending offers is less than concurrency: false","timestamp":"2016-06-16T12:29:55.184Z"}
{"level":"debug","message":"negotiator returned: true","timestamp":"2016-06-16T12:29:55.184Z"}
{"level":"debug","message":"we have enough free space: true","timestamp":"2016-06-16T12:29:55.184Z"}
{"level":"debug","message":"Not taking the contract...","timestamp":"2016-06-16T12:29:55.184Z"}
{"level":"info","message":"sending PUBLISH message to {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080195184}","timestamp":"2016-06-16T12:29:55.184Z"}
{"level":"debug","message":"queuing callback for reponse to 631647fca562d72cd8c30cf8b7d9a9d6222e78bc","timestamp":"2016-06-16T12:29:55.247Z"}
{"level":"info","message":"sending PUBLISH message to {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6155,\"nodeID\":\"2b6b9175dcbdcbbce3337ca9e7a73946b82452ea\",\"lastSeen\":1466080195248}","timestamp":"2016-06-16T12:29:55.248Z"}
{"level":"debug","message":"queuing callback for reponse to 9531b7a6910a76b3d3e0f1f13dfc9b5b11e97ce5","timestamp":"2016-06-16T12:29:55.282Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6000,\"nodeID\":\"6ad8a5a30528b3572f3e80cbd5a3eb4c52b8b8ac\",\"lastSeen\":1466080195354}","timestamp":"2016-06-16T12:29:55.354Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6000,\"nodeID\":\"6ad8a5a30528b3572f3e80cbd5a3eb4c52b8b8ac\",\"lastSeen\":1466080195354}","timestamp":"2016-06-16T12:29:55.354Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.354Z"}
{"level":"debug","message":"checking pending rpc callback stack for 4d22b95e38fe5579330e841715a5242d22c7f254","timestamp":"2016-06-16T12:29:55.423Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"veeit30163.dyndns.org\",\"port\":6000,\"nodeID\":\"6ad8a5a30528b3572f3e80cbd5a3eb4c52b8b8ac\",\"lastSeen\":1466080194386}","timestamp":"2016-06-16T12:29:55.424Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.424Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080195426}","timestamp":"2016-06-16T12:29:55.426Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080195426}","timestamp":"2016-06-16T12:29:55.426Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.426Z"}
{"level":"debug","message":"checking pending rpc callback stack for a111b7dd674abfc4950fd29eac0b95bebb721f38","timestamp":"2016-06-16T12:29:55.493Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"symbiont.spdns.de\",\"port\":6050,\"nodeID\":\"ef409f8911a486759f59276ad83626bdb0fd97d4\",\"lastSeen\":1466080194481}","timestamp":"2016-06-16T12:29:55.493Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.493Z"}
{"level":"info","message":"received valid message from {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080195580}","timestamp":"2016-06-16T12:29:55.580Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080195580}","timestamp":"2016-06-16T12:29:55.580Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.580Z"}
{"level":"debug","message":"checking pending rpc callback stack for 6fded0734b113d64cea6a0e1864fbc1593338b52","timestamp":"2016-06-16T12:29:55.650Z"}
{"level":"debug","message":"updating contact {\"protocol\":\"0.7.0\",\"address\":\"server018.storj.dk\",\"port\":5018,\"nodeID\":\"1495a38adfec7117a2dc84135be8ea9612841a86\",\"lastSeen\":1466080194420}","timestamp":"2016-06-16T12:29:55.651Z"}
{"level":"debug","message":"contact already in bucket, moving to tail","timestamp":"2016-06-16T12:29:55.651Z"}
{"level":"debug","message":"finished iteration, handling results","timestamp":"2016-06-16T12:29:55.651Z"}
{"level":"debug","message":"shortlist is full or there are no known nodes closer to key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:55.651Z"}
{"level":"debug","message":"found 48 nodes close to key c22b9d7cbd82924e031197ebf806ae26a0237e9d","timestamp":"2016-06-16T12:29:55.652Z"}
{"level":"warn","message":"Could not locate renter for offer","timestamp":"2016-06-16T12:29:55.652Z"}
```

### Steps to Reproduce

Please include the steps the reproduce the issue, numbered below. Include as
much detail as possible.

1. Start node with MAX_CONCURRENT_OFFERS = 1
2. Wait a few hours and search for `Could not locate renter for offer`
3. Count the number of `sending FIND_NODE` for this offer. You will find 3 of them and that is insufficient.

**Comment by bookchin** [comment]
This is another kad issue. Not really a bug, but rather our usage of `Router#findNode` is inconsistent with what kad expects. Same issue as before, we are performing the network walk by searching for the SHA1 of the nodeID instead of the nodeID itself, which will end up asking the wrong neighbors for the wrong thing.

I will add the option to kad to perform a strict, non-coercive lookup and update our usage in core.

**Comment by bookchin** [comment]
Updated `kad@1.5.13`, should fix the issue. Can you give it go?

**Comment by bookchin** [comment]
you can install `bookchin/storj-core` for the other fixes in #183 if you like
